include/mbgl/storage/default_file_source.hpp

```diff
@@ -18,6 +144,49 @@ class DefaultFileSource : public FileSource {
 
     std::unique_ptr<FileRequest> request(const Resource&, Callback) override;
 
+    /*
+     * Retrieve all regions in the offline database.
+     *
+     * The query will be executed asynchronously and the results passed to the given
+     * callback, which will be executed on the database thread; it is the responsibility
+     * of the SDK bindings to re-execute a user-provided callback on the main thread.
+     */
+    void listOfflineRegions(std::function<void (std::exception_ptr,
+                                                optional<std::vector<OfflineRegion>>)>);
+
+    /*
+     * Create an offline region in the database.
+     *
+     * When the initial database queries have completed, the provided callback will be
+     * executed on the database thread; it is the responsibility of the SDK bindings
+     * to re-execute a user-provided callback on the main thread.
+     *
+     * Note that the resulting region will be in an inactive download state; to begin
+     * downloading resources, call `setDownloadState(OfflineRegionDownloadState::Active)`,
+     * optionally registering an `OfflineRegionObserver` beforehand.
+     */
+    void createOfflineRegion(const OfflineRegionDefinition& definition,
```

Comment:
Updates will not be part of this API, at least in the first version. With deduplication, you can get much the same behavior by creating a new region and then deleting the old one.